lib/initial-setup/WP_Auth0_InitialSetup_Consent.php

```diff
@@ -128,19 +144,22 @@ public function consent_callback( $name ) {
 		$connection_exists     = false;
 		$connection_pwd_policy = null;
 
-		$connections = WP_Auth0_Api_Client::search_connection( $domain, $app_token );
+		$connections = WP_Auth0_Api_Client::search_connection( $domain, $app_token, 'auth0' );
 
-		foreach ( $connections as $connection ) {
+		if ( $should_create_and_update_connection && ! empty( $connections ) && is_array( $connections ) ) {
+			foreach ( $connections as $connection ) {
 
-			if ( in_array( $client_id, $connection->enabled_clients ) ) {
-				if ( $connection->strategy === 'auth0' && $should_create_and_update_connection ) {
+				if ( in_array( $client_id, $connection->enabled_clients ) ) {
 
 					if ( $db_connection_name === $connection->name ) {
-						$connection_exists     = $connection->id;
-						$connection_pwd_policy = ( isset( $connection->options ) && isset( $connection->options->passwordPolicy ) ) ? $connection->options->passwordPolicy : null;
+						$connection_exists = $connection->id;
+						if ( isset( $connection->options ) && isset( $connection->options->passwordPolicy ) ) {
+							$connection_pwd_policy = $connection->options->passwordPolicy;
+						}
 					} else {
-						$enabled_clients = array_diff( $connection->enabled_clients, array( $client_id ) );
-						WP_Auth0_Api_Client::update_connection( $domain, $app_token, $connection->id, array( 'enabled_clients' => array_values( $enabled_clients ) ) );
+						$u_connection                  = clone $connection;
```

Comment:
Add a comment above this line: "//disable connection for this application/client id"